lib/auth/clt.go

```diff
@@ -821,6 +821,18 @@ func (c *Client) AuthenticateWebUser(ctx context.Context, req AuthenticateUserRe
 // AuthenticateSSHUser authenticates SSH console user, creates and  returns a pair of signed TLS and SSH
 // short lived certificates as a result
 func (c *Client) AuthenticateSSHUser(ctx context.Context, req AuthenticateSSHRequest) (*SSHLoginResponse, error) {
+	if req.HeadlessAuthenticationID != "" {
+		// Replace the client timeout with the default callback timeout for this request.
+		previousResponseHeaderTimeout := c.HTTPClient.transport.ResponseHeaderTimeout
+		previousClientTimeout := c.HTTPClient.HTTPClient().Timeout
+		c.HTTPClient.transport.ResponseHeaderTimeout = defaults.CallbackTimeout
+		c.HTTPClient.HTTPClient().Timeout = defaults.CallbackTimeout
+		defer func() {
+			c.HTTPClient.transport.ResponseHeaderTimeout = previousResponseHeaderTimeout
+			c.HTTPClient.HTTPClient().Timeout = previousClientTimeout
+		}()
```

Comment:
Unfortunately there isn't an easy way to do this since the timeout is only configurable at the client level, not the request level, and the proxy api server uses the same auth client for all requests. I will try handling this by adding a `HTTPClient.WithClientParams(...roudtrip.ClientParams)` to create a clone of the HTTP client with the given params applied.

Reply:
@rosstimothy I did the above in https://github.com/gravitational/teleport/pull/22726/commits/403b158c3d9b39d137757c17a1a987ea4889e6ed

I moved the http client out of `lib/auth/clt.go` and moved some code around to improve the code quality (Adding HTTPClientConfig, CheckAndSetDefaults, etc.), but didn't make any functional changes, other than adding `HTTPClient.Clone()`.